source/DataElement/AMDSFlatArray.cpp

```diff
@@ -399,6 +399,67 @@ bool AMDSFlatArray::setValue(int index, double  value)
 	return true;
 }
 
+const QVector<double> AMDSFlatArray::asConstVectorDouble() const
+{
+	QVector<double> targetVector;
+	switch(dataType_){
+	case AMDSDataTypeDefinitions::Signed8:
+		foreach(qint8 value, vectorQint8_) {
+			targetVector.append((double)value);
```

Comment:
This needs to be changed right away!  We can not append to QVectors like this.  Every append causes QVector to reallocate all its memory and is _EXTREMELY_ inefficient.  These need to be changed to:

```c++
QVector<double> targetVector = QVector<double>(vectorQint8_.size());

for (int i = 0, size = targetVector.size(); i < size; i++)
     targetVector[i] = double(vectorQint8_.at(i));
```